src/sentry/snuba/metrics/fields/snql.py

```diff
@@ -367,3 +369,62 @@ def count_web_vitals_snql_factory(aggregate_filter, org_id, measurement_rating,
         ],
         alias=alias,
     )
+
+
+def count_transaction_with_condition_snql_factory(aggregate_filter, org_id, condition, alias=None):
+    is_unparameterized = "is_unparameterized"
+    is_null = "is_null"
+    has_value = "has_value"
+
+    def _generate_transaction_tag_value_filter(operation, inner_condition):
+        if inner_condition == is_unparameterized:
+            inner_tag_value = resolve_tag_value(
+                UseCaseKey.PERFORMANCE, org_id, "<< unparameterized >>"
+            )
+        elif inner_condition == is_null:
+            inner_tag_value = (
+                "" if options.get("sentry-metrics.performance.tags-values-are-strings") else 0
+            )
+        else:
+            raise InvalidParams("invalid condition for tag value filter")
+
+        return Function(
+            operation,
+            [
+                Column(
+                    resolve_tag_key(
+                        UseCaseKey.PERFORMANCE,
+                        org_id,
+                        "transaction",
+                    )
+                ),
+                inner_tag_value,
+            ],
+        )
+
+    if condition in [is_unparameterized, is_null]:
+        snql_function = _generate_transaction_tag_value_filter("equals", condition)
```

Comment:
I think that we should drop the `_` convention that we use for inner methods (I saw it in other `snql` factories) because actually from Python's semantics the use of `_` should be done only on methods and variables that can be accessed from the outside but inner functions are only accessible within a specific function scope.

Reply:
Which other functions are you referring to? Can you provide an example?

Reply:
`def _aggregation_on_tx_status_func_factory(aggregate):
    def _get_snql_conditions(org_id, metric_ids, exclude_tx_statuses):
        metric_match = Function("in", [Column("metric_id"), list(metric_ids)])
        assert exclude_tx_statuses is not None
        if len(exclude_tx_statuses) == 0:
            return metric_match`

Reply:
https://github.com/getsentry/sentry/blob/fbe83d2e84865111c8f595f2ec0273180a12b261/src/sentry/snuba/metrics/fields/snql.py#L66-L66

Reply:
agreed lets fix that in subsequent PRs